tools/shuffle_tfrecords_beam.py

```diff
@@ -159,12 +159,21 @@ def sha1(input_bytes):
           | 'DropKey' >> beam.FlatMap(lambda x: x[1]))
 
 
-def make_config_string(name, tfrecord_path, num_examples):
-  return textwrap.dedent("""
-  name: "{}"
-  tfrecord_path: "{}-?????-of-?????.tfrecord.gz"
-  num_examples: {}
-  """.format(name, tfrecord_path, num_examples))
+# SparkRunner has some issue with using lambda functions in beam.Map
+# Hence, this has been turned into a callable
+class MakeConfigStringCaller:
+  def __init__(self, dataset_name, output_pattern_prefix):
+    self.name = dataset_name
+    self.tfrecord_path = output_pattern_prefix
+
+  def __call__(self, num_examples):
```

Comment:
The approach I've usually seen is defining a class that inherits from beam.DoFn. For example:

```
class MakeConfigStringCaller(beam.DoFn):

  def __init__(self, ...):
     ...

  def process(self, num_examples):
    ...
```

Usage:

```
config_str = (
  num_examples 
  | 'MakeConfigStr' >> beam.ParDo(MakeConfigStringCaller(dataset_name, output_pattern_prefix)))
```

Any reason to not use this approach? It's pretty similar to what you have, but is a Beam pattern I've seen used more commonly than a callable. I'm not sure which of these approaches would be faster though.